lib/core/module/presentation/widgets/module_card.dart

```diff
@@ -0,0 +1,177 @@
+import '../../domain/module.dart';
+import '../../../constants/color_constants.dart';
+import '../../../constants/widget_constants.dart';
+
+const double width = 214.0;
+const double height = 612.0;
+
+@immutable
+class ModuleCard extends StatelessWidget {
+  final Module module;
+
+  ModuleCard(this.module);
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      width: width,
+      height: height,
+      decoration: BoxDecoration(
+        color: Theme.of(context).colorScheme.surface,
+        border: Border.all(
+          color: kBorderColor,
+        ),
+        borderRadius: BorderRadius.all(Radius.circular(kSmallCornerRadius)),
+      ),
+      child: Padding(
+        padding: const EdgeInsets.all(16.0),
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            Row(
+              children: [
+                Expanded(
+                  child: Text(
+                    module.code.code,
+                    style: Theme.of(context).textTheme.bodyText1,
+                  ),
+                ),
+                SizedBox(
+                  width: 22,
+                  child: IconButton(
+                    icon: Icon(FluentIcons.more_vertical_20_regular),
+                    color: Colors.red,
+                    // TODO: Change this later.
+                    onPressed: () {},
+                  ),
+                ),
+              ],
+            ),
+            Text(
+              module.name,
+              style: Theme.of(context).textTheme.headline6,
+            ),
+            Expanded(
+              child: Row(
+                crossAxisAlignment: CrossAxisAlignment.end,
+                mainAxisAlignment: MainAxisAlignment.end,
+                children: [
+                  Text(
+                    getFormattedCardFooter(),
+                    textAlign: TextAlign.right,
+                    style: Theme.of(context).textTheme.caption,
+                  ),
+                ],
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  // Formats such that it's something like Y1 · Sem 2 · Enrichment.
+  String getFormattedCardFooter() =>
+      module.levels[0].getFormattedLevels(module.levels) +
+      ' · ' +
+      module.semesterType.getFormattedSemesterType +
+      ' · ' +
+      module.moduleType.getFormattedModuleType;
+}
+
+extension LevelExtension on Level {
+  String get level => describeEnum(this);
+
+  String get getFormattedLevel {
+    switch (this) {
+      case Level.year1:
+        return '1';
+      case Level.year2:
+        return '2';
+      case Level.year3:
+        return '3';
+      case Level.year4:
+        return '4';
+      case Level.year5:
+        return '5';
+      case Level.year6:
+        return '6';
+      default:
+        return '';
+    }
+  }
+
+  String getFormattedLevels(List<Level> levels) {
+    levels.sort((level1, level2) =>
```

Comment:
Not a good idea to mutate inputs. This can lead to unexpected results, perhaps clone the list of levels first